internal/rules_test.go

```diff
@@ -210,3 +210,48 @@ func TestToGraphiteName_Error(t *testing.T) {
 	checkConversionErrorCode(t, err, CannotInterpolate)
 	a.EqString(string(reversed), "")
 }
+
+func Test_interpolateTags(t *testing.T) {
+
+	for _, testCase := range []struct {
+		pattern string
+		tagSet  api.TagSet
+		enforce bool
+		result  string
+	}{
+		// note that the result <fail> indicates that the test case should fail to parse
```

Comment:
Should use another flag to indicate whether the test should succeed or fail instead of a special `<fail>` string